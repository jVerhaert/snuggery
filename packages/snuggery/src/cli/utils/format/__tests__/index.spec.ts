import {formatMarkdownish} from '../index';

function reverse(text: string): string {
	return Array.from(text).reverse().join('');
}

describe('formatMarkdownish', () => {
	const texts = [
		'lorem',

		// should collapse
		'lorem\nipsum',

		// should not collapse
		'lorem  \nipsum',
		'lorem\n\nipsum',

		// Lists

		'- lorem\n- ipsum\n- dolor',
		'- lorem ipsum dolor sit amet, the quick brown fox jumps over the lazy dog\n- ipsum\n- dolor sit amet, the lazy dog is jumped over by the quick brown fox',

		// Code
		'`a code` block',
		"what happens when a code block is split over multiple lines? `let's find out shan't we?`",

		// Bold
		'**bold** text',
		"what happens when bold text is split over multiple lines? **let's find out shan't we?**",

		// Pre-indented (spaces & tabs)
		`
      This text is already
      indented because it's
      multiline and part of
      a section of real code.
    `,
		`
			This text is already
			indented because it's
			multiline and part of
			a section of real code.
		`,
	];

	it('should format correctly', () => {
		for (const text of texts) {
			expect(formatMarkdownish(text)).toMatchSnapshot();
		}
	});

	it('should format correctly with infinite line length', () => {
		for (const text of texts) {
			expect(
				formatMarkdownish(text, {maxLineLength: Infinity}),
			).toMatchSnapshot();
		}
	});

	it('should format correctly with indentation', () => {
		for (const text of texts) {
			expect(formatMarkdownish(text, {indentation: 2})).toMatchSnapshot();
			expect(formatMarkdownish(text, {indentation: 20})).toMatchSnapshot();
		}
	});

	it('should format correctly with format', () => {
		for (const text of texts) {
			expect(
				formatMarkdownish(text, {
					format: {
						bold: reverse,
					},
				}),
			).toMatchSnapshot();

			expect(
				formatMarkdownish(text, {
					format: {
						code: reverse,
					},
				}),
			).toMatchSnapshot();
		}
	});
});
