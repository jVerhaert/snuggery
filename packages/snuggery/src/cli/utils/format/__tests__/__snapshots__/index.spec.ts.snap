// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatMarkdownish should format correctly 1`] = `"lorem"`;

exports[`formatMarkdownish should format correctly 2`] = `"lorem ipsum"`;

exports[`formatMarkdownish should format correctly 3`] = `
"lorem
ipsum"
`;

exports[`formatMarkdownish should format correctly 4`] = `
"lorem

ipsum"
`;

exports[`formatMarkdownish should format correctly 5`] = `
"- lorem
- ipsum
- dolor"
`;

exports[`formatMarkdownish should format correctly 6`] = `
"- lorem ipsum dolor sit amet, the quick brown fox jumps over the lazy dog
- ipsum
- dolor sit amet, the lazy dog is jumped over by the quick brown fox"
`;

exports[`formatMarkdownish should format correctly 7`] = `"\`a code\` block"`;

exports[`formatMarkdownish should format correctly 8`] = `
"what happens when a code block is split over multiple lines? \`let's find out
shan't we?\`"
`;

exports[`formatMarkdownish should format correctly 9`] = `"**bold** text"`;

exports[`formatMarkdownish should format correctly 10`] = `
"what happens when bold text is split over multiple lines? **let's find out
shan't we?**"
`;

exports[`formatMarkdownish should format correctly 11`] = `
"This text is already indented because it's multiline and part of a section of
real code."
`;

exports[`formatMarkdownish should format correctly 12`] = `
"This text is already indented because it's multiline and part of a section of
real code."
`;

exports[`formatMarkdownish should format correctly with format 1`] = `"lorem"`;

exports[`formatMarkdownish should format correctly with format 2`] = `"lorem"`;

exports[`formatMarkdownish should format correctly with format 3`] = `"lorem ipsum"`;

exports[`formatMarkdownish should format correctly with format 4`] = `"lorem ipsum"`;

exports[`formatMarkdownish should format correctly with format 5`] = `
"lorem
ipsum"
`;

exports[`formatMarkdownish should format correctly with format 6`] = `
"lorem
ipsum"
`;

exports[`formatMarkdownish should format correctly with format 7`] = `
"lorem

ipsum"
`;

exports[`formatMarkdownish should format correctly with format 8`] = `
"lorem

ipsum"
`;

exports[`formatMarkdownish should format correctly with format 9`] = `
"- lorem
- ipsum
- dolor"
`;

exports[`formatMarkdownish should format correctly with format 10`] = `
"- lorem
- ipsum
- dolor"
`;

exports[`formatMarkdownish should format correctly with format 11`] = `
"- lorem ipsum dolor sit amet, the quick brown fox jumps over the lazy dog
- ipsum
- dolor sit amet, the lazy dog is jumped over by the quick brown fox"
`;

exports[`formatMarkdownish should format correctly with format 12`] = `
"- lorem ipsum dolor sit amet, the quick brown fox jumps over the lazy dog
- ipsum
- dolor sit amet, the lazy dog is jumped over by the quick brown fox"
`;

exports[`formatMarkdownish should format correctly with format 13`] = `"\`a code\` block"`;

exports[`formatMarkdownish should format correctly with format 14`] = `"\`edoc a\` block"`;

exports[`formatMarkdownish should format correctly with format 15`] = `
"what happens when a code block is split over multiple lines? \`let's find out
shan't we?\`"
`;

exports[`formatMarkdownish should format correctly with format 16`] = `
"what happens when a code block is split over multiple lines? \`?ew t'nahs
tuo dnif s'tel\`"
`;

exports[`formatMarkdownish should format correctly with format 17`] = `"**dlob** text"`;

exports[`formatMarkdownish should format correctly with format 18`] = `"**bold** text"`;

exports[`formatMarkdownish should format correctly with format 19`] = `
"what happens when bold text is split over multiple lines? **?ew t'nahs
tuo dnif s'tel**"
`;

exports[`formatMarkdownish should format correctly with format 20`] = `
"what happens when bold text is split over multiple lines? **let's find out
shan't we?**"
`;

exports[`formatMarkdownish should format correctly with format 21`] = `
"This text is already indented because it's multiline and part of a section of
real code."
`;

exports[`formatMarkdownish should format correctly with format 22`] = `
"This text is already indented because it's multiline and part of a section of
real code."
`;

exports[`formatMarkdownish should format correctly with format 23`] = `
"This text is already indented because it's multiline and part of a section of
real code."
`;

exports[`formatMarkdownish should format correctly with format 24`] = `
"This text is already indented because it's multiline and part of a section of
real code."
`;

exports[`formatMarkdownish should format correctly with indentation 1`] = `"  lorem"`;

exports[`formatMarkdownish should format correctly with indentation 2`] = `"                    lorem"`;

exports[`formatMarkdownish should format correctly with indentation 3`] = `"  lorem ipsum"`;

exports[`formatMarkdownish should format correctly with indentation 4`] = `"                    lorem ipsum"`;

exports[`formatMarkdownish should format correctly with indentation 5`] = `
"  lorem
  ipsum"
`;

exports[`formatMarkdownish should format correctly with indentation 6`] = `
"                    lorem
                    ipsum"
`;

exports[`formatMarkdownish should format correctly with indentation 7`] = `
"  lorem
  
  ipsum"
`;

exports[`formatMarkdownish should format correctly with indentation 8`] = `
"                    lorem
                    
                    ipsum"
`;

exports[`formatMarkdownish should format correctly with indentation 9`] = `
"  - lorem
  - ipsum
  - dolor"
`;

exports[`formatMarkdownish should format correctly with indentation 10`] = `
"                    - lorem
                    - ipsum
                    - dolor"
`;

exports[`formatMarkdownish should format correctly with indentation 11`] = `
"  - lorem ipsum dolor sit amet, the quick brown fox jumps over the lazy dog
  - ipsum
  - dolor sit amet, the lazy dog is jumped over by the quick brown fox"
`;

exports[`formatMarkdownish should format correctly with indentation 12`] = `
"                    - lorem ipsum dolor sit amet, the quick brown fox jumps over
                      the lazy dog
                    - ipsum
                    - dolor sit amet, the lazy dog is jumped over by the quick
                      brown fox"
`;

exports[`formatMarkdownish should format correctly with indentation 13`] = `"  \`a code\` block"`;

exports[`formatMarkdownish should format correctly with indentation 14`] = `"                    \`a code\` block"`;

exports[`formatMarkdownish should format correctly with indentation 15`] = `
"  what happens when a code block is split over multiple lines? \`let's find out
  shan't we?\`"
`;

exports[`formatMarkdownish should format correctly with indentation 16`] = `
"                    what happens when a code block is split over multiple lines?
                    \`let's find out shan't we?\`"
`;

exports[`formatMarkdownish should format correctly with indentation 17`] = `"  **bold** text"`;

exports[`formatMarkdownish should format correctly with indentation 18`] = `"                    **bold** text"`;

exports[`formatMarkdownish should format correctly with indentation 19`] = `
"  what happens when bold text is split over multiple lines? **let's find out
  shan't we?**"
`;

exports[`formatMarkdownish should format correctly with indentation 20`] = `
"                    what happens when bold text is split over multiple lines?
                    **let's find out shan't we?**"
`;

exports[`formatMarkdownish should format correctly with indentation 21`] = `
"  This text is already indented because it's multiline and part of a section of
  real code."
`;

exports[`formatMarkdownish should format correctly with indentation 22`] = `
"                    This text is already indented because it's multiline and
                    part of a section of real code."
`;

exports[`formatMarkdownish should format correctly with indentation 23`] = `
"  This text is already indented because it's multiline and part of a section of
  real code."
`;

exports[`formatMarkdownish should format correctly with indentation 24`] = `
"                    This text is already indented because it's multiline and
                    part of a section of real code."
`;

exports[`formatMarkdownish should format correctly with infinite line length 1`] = `"lorem"`;

exports[`formatMarkdownish should format correctly with infinite line length 2`] = `"lorem ipsum"`;

exports[`formatMarkdownish should format correctly with infinite line length 3`] = `
"lorem
ipsum"
`;

exports[`formatMarkdownish should format correctly with infinite line length 4`] = `
"lorem

ipsum"
`;

exports[`formatMarkdownish should format correctly with infinite line length 5`] = `
"- lorem
- ipsum
- dolor"
`;

exports[`formatMarkdownish should format correctly with infinite line length 6`] = `
"- lorem ipsum dolor sit amet, the quick brown fox jumps over the lazy dog
- ipsum
- dolor sit amet, the lazy dog is jumped over by the quick brown fox"
`;

exports[`formatMarkdownish should format correctly with infinite line length 7`] = `"\`a code\` block"`;

exports[`formatMarkdownish should format correctly with infinite line length 8`] = `"what happens when a code block is split over multiple lines? \`let's find out shan't we?\`"`;

exports[`formatMarkdownish should format correctly with infinite line length 9`] = `"**bold** text"`;

exports[`formatMarkdownish should format correctly with infinite line length 10`] = `"what happens when bold text is split over multiple lines? **let's find out shan't we?**"`;

exports[`formatMarkdownish should format correctly with infinite line length 11`] = `"This text is already indented because it's multiline and part of a section of real code."`;

exports[`formatMarkdownish should format correctly with infinite line length 12`] = `"This text is already indented because it's multiline and part of a section of real code."`;
